parsec/backend/asgi/rpc.py

```diff
@@ -63,15 +93,194 @@ async def anonymous_api(raw_organization_id: str):
         except OrganizationAlreadyExistsError:
             pass
 
-    # Retreive command
+    # Retrieve command
     client_ctx = AnonymousClientContext(organization_id)
     try:
         if not isinstance(cmd, str):
-            raise KeyError()
+            raise KeyError(f"`cmd` type is not a string but a {type(cmd).__name__}")
         cmd_func = backend.apis[ClientType.ANONYMOUS][cmd]
-    except KeyError:
+    except KeyError as e:
+        logger.warning(f"Anonymous command {cmd} doesn't exist ({e})")
         return rpc_msgpack_rep({"status": "unknown_command"})
 
     # Run command
     rep = await cmd_func(client_ctx, msg)
     return rpc_msgpack_rep(rep)
+
+
+@rpc_blueprint.route("/authenticated/<raw_organization_id>", methods=["POST"])
+async def authenticated_api(raw_organization_id: str):
+    backend: BackendApp = g.backend
+    logger = current_app.logger.getChild(authenticated_api.__name__)
+
+    logger.debug(
+        "Received call in the authenticated api for the organization {org_id} from the device={dev_id}".format(
+            org_id=raw_organization_id, dev_id=request.headers.get("Author", "Missing")
+        )
+    )
+    try:
+        organization_id = OrganizationID(raw_organization_id)
+    except ValueError:
+        logger.warning(f"Invalid organization id: {raw_organization_id}")
+        abort(404)
+
+    try:
+        organization = await backend.organization.get(organization_id)
+    except OrganizationNotFoundError:
+        logger.warning(f"Organization {organization_id} not found")
+        abort(400)
+    else:
+        if organization.is_expired:
+            logger.warning(f"Organization {organization_id} is expired")
+            abort(400)
+
+    body: bytes = await request.get_data(cache=False)
+    try:
+        user, device = await verify_auth_request(body, request.headers, organization, backend)
+    except (HttpProtocolError, UserNotFoundError) as e:
+        logger.warning(f"Invalid request, cause: {e}")
+        abort(400)
+    except AuthenticationProtocolError as e:
+        logger.warning(f"Failed to authenticate request because of {e}")
+        abort(401)
+
+    api_version = get_api_version(request.headers)
+
+    client_ctx = build_client_context(organization, user, device, api_version)
+
+    # Unpack the verified body
+    if request.headers["Content-Type"] != "application/msgpack":
```

Comment:
`"application/msgpack"` should be a constant at the top of the file

Reply:
done